src/ucp/proto/proto_am.inl

```diff
@@ -596,11 +596,15 @@ ucp_proto_get_short_max(const ucp_request_t *req,
 
 static UCS_F_ALWAYS_INLINE int
 ucp_proto_is_inline(ucp_ep_h ep, const ucp_memtype_thresh_t *max_eager_short,
-                    ssize_t length)
+                    ssize_t length, ucs_memory_type_t mem_type)
 {
+    if (ucs_likely(mem_type == UCS_MEMORY_TYPE_UNKNOWN) &&
```

Comment:
this PR adds at least 2 extra branches on the fast path: `(param->op_attr_mask & UCP_OP_ATTR_FIELD_MEMORY_TYPE)` and this one.
We should make sure these branches are executed only when `ucs_memtype_cache_is_empty()` is false (i.e. there is GPU in the system)

Reply:
What if there is GPU in the system and GPU buffers are passed for communications, but the user always provides the memory type? In that case, memtype cache will be empty even though we're using GPU buffers, isn't that so?

Reply:
no, the mem type cache will not be empty because UCM code adds all "unknown" regions to it during initialization.
and anyway, we would add any cuda memory to the memtype cache when it's allocated.

Reply:
@yosefe I updated the code and reduced it to one extra branch.